checker/src/main/java/alluxio/checker/HiveIntegrationChecker.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.checker;
+
+import com.beust.jcommander.IStringConverter;
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParameterException;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * Some Hive queries to test the integration of Hive with Alluxio.
+ * Supports two options: Alluxio is configured as Hive default filesytem
+ * and Alluxio is used as a location of Hive tables other than Hive default filesystem.
+ *
+ * This checker requires a running Hive cluster and a running Alluxio cluster.
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized
+ * in Hive queries.
+ */
+@SuppressFBWarnings("SQL_PREPARED_STATEMENT_GENERATED_FROM_NONCONSTANT_STRING")
+public class HiveIntegrationChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(HiveIntegrationChecker.class);
+  private static final String FAIL_TO_FIND_CLASS_MESSAGE = "Please distribute "
+      + "the Alluxio client jar on the classpath of the application across different nodes.\n\n"
+      + "Please set HIVE_AUX_JARS_PATH either in shell or in conf/hive-env.sh "
+      + "before starting hiveserver2.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Running-Hive-with-Alluxio.html\n";
+  private static final String FAIL_TO_FIND_FS_MESSAGE = "Please check the fs.alluxio.impl property "
+      + "in core-site.xml file of your Hadoop installation.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Debugging-Guide.html\n";
+  private static final String TEST_FAILED_MESSAGE = "***** Integration test failed. *****\n";
+  private static final String TEST_PASSED_MESSAGE = "***** Integration test passed. *****\n";
+
+  /** Converts input string USER_MODE to enum Mode.*/
+  public static class ModeConverter implements IStringConverter<Mode> {
+    @Override
+    public Mode convert(String value) {
+      if (value.equalsIgnoreCase("dfs")) {
```

Comment:
why use equalsIgnoreCase?

Reply:
In command line interface, we seem to have the tradition that all the arguments are in lower case. I want to support that no matter users provide dfs or DFS as UserMode, we are able to transfer it to MODE(DFS/LOCATION).

Reply:
case-insensitivity is both good and bad. It can lead to inconsistent usage where some people use "dfs", some use "Dfs", some use "DFS", and nobody knows which way is the "proper" way to do it. Since the rest of the shell (and shells in general) are case-sensitive, I think we should probably follow suit here

Reply:
then I only support dfs and location here to make it consistent with the shell usage.